node_modules/oae-activity/tests/test-email.js

```diff
@@ -855,6 +857,59 @@ describe('Activity Email', function() {
     });
 
     /**
+     * Test that verifies that activity emails are not delivered to users who have deleted their
+     * profiles
+     */
+    it('verify email is not delivered to users who have since deleted their profile', function(callback) {
+        TestsUtil.generateTestUsers(camAdminRestContext, 2, function(err, users, mrvisser, simong) {
+            assert.ok(!err);
+
+            // Set the appropriate email preference for mrvisser
+            RestAPI.User.updateUser(mrvisser.restContext, mrvisser.user.id, {'emailPreference': 'immediate'}, function(err) {
+                assert.ok(!err);
+
+                // Generate an email activity for mrvisser
+                RestAPI.Content.createLink(simong.restContext, 'Google', 'Awesome Google', 'public', 'http://www.google.ca', [mrvisser.user.id], [], [], function(err, link) {
+                    assert.ok(!err);
+
+                    // Ensure activities and notifications get delivered to mrvisser
+                    MqTestUtil.whenTasksEmpty(ActivityConstants.mq.TASK_ACTIVITY, function() {
+                        ActivityNotifications.whenNotificationsEmpty(function() {
+
+                            // Delete mrvisser now with pending emails in their notifications feed
+                            PrincipalsTestUtil.assertDeleteUserSucceeds(camAdminRestContext, camAdminRestContext, mrvisser.user.id, function() {
+
+                                // Deliver the activity and collect all emails associated to it,
+                                // ensuring no emails actually get delivered
+                                EmailTestUtil.collectAndFetchEmailsForBucket(0, 'immediate', null, null, function(messages) {
+                                    assert.strictEqual(messages.length, 0);
+
+                                    // Restore mrvisser
+                                    PrincipalsTestUtil.assertRestoreUserSucceeds(camAdminRestContext, mrvisser.user.id, function() {
+
+                                        // Simong will trigger another activity, annoying-sauce
+                                        RestAPI.Content.createLink(simong.restContext, 'Google', 'Awesome Google', 'public', 'http://www.google.ca', [mrvisser.user.id], [], [], function(err, link) {
+                                            assert.ok(!err);
+
+                                            // Collect and ensure mrvisser now gets the email for the latest activity
+                                            EmailTestUtil.collectAndFetchEmailsForBucket(0, 'immediate', null, null, function(messages) {
+                                                assert.strictEqual(messages.length, 1);
+                                                assert.strictEqual(messages[0].to[0].address, mrvisser.user.email);
```

Comment:
It will contain the initial one because it aggregates in the feed. But I can make sure it contains both.